
  0%|                                                                                                                                      | 0/504 [00:00<?, ?it/s]
  0%|▏                                                                                                                           | 1/504 [00:14<1:59:37, 14.27s/it]Traceback (most recent call last):
  File "/scratch/project/dd-21-23/test/gpt-neo-fine-tuning-example/./gpt_j_deepspeed.py", line 74, in <module>
    'labels': torch.stack([f[0] for f in data])}).train()
  File "/home/vyomkeshj/.conda/envs/ml_test/lib/python3.10/site-packages/transformers/trainer.py", line 1325, in train
    tr_loss_step = self.training_step(model, inputs)
  File "/home/vyomkeshj/.conda/envs/ml_test/lib/python3.10/site-packages/transformers/trainer.py", line 1884, in training_step
    loss = self.compute_loss(model, inputs)
  File "/home/vyomkeshj/.conda/envs/ml_test/lib/python3.10/site-packages/transformers/trainer.py", line 1916, in compute_loss
    outputs = model(**inputs)
  File "/home/vyomkeshj/.conda/envs/ml_test/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1110, in _call_impl
    return forward_call(*input, **kwargs)
  File "/home/vyomkeshj/.conda/envs/ml_test/lib/python3.10/site-packages/deepspeed/utils/nvtx.py", line 11, in wrapped_fn
    return func(*args, **kwargs)
  File "/home/vyomkeshj/.conda/envs/ml_test/lib/python3.10/site-packages/deepspeed/runtime/engine.py", line 1616, in forward
    loss = self.module(*inputs, **kwargs)
  File "/home/vyomkeshj/.conda/envs/ml_test/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1128, in _call_impl
    result = forward_call(*input, **kwargs)
  File "/home/vyomkeshj/.conda/envs/ml_test/lib/python3.10/site-packages/transformers/models/gptj/modeling_gptj.py", line 782, in forward
    transformer_outputs = self.transformer(
  File "/home/vyomkeshj/.conda/envs/ml_test/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1128, in _call_impl
    result = forward_call(*input, **kwargs)
  File "/home/vyomkeshj/.conda/envs/ml_test/lib/python3.10/site-packages/transformers/models/gptj/modeling_gptj.py", line 636, in forward
    outputs = block(
  File "/home/vyomkeshj/.conda/envs/ml_test/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1128, in _call_impl
    result = forward_call(*input, **kwargs)
  File "/home/vyomkeshj/.conda/envs/ml_test/lib/python3.10/site-packages/transformers/models/gptj/modeling_gptj.py", line 280, in forward
    attn_outputs = self.attn(
  File "/home/vyomkeshj/.conda/envs/ml_test/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1128, in _call_impl
    result = forward_call(*input, **kwargs)
  File "/home/vyomkeshj/.conda/envs/ml_test/lib/python3.10/site-packages/transformers/models/gptj/modeling_gptj.py", line 229, in forward
    attn_output, attn_weights = self._attn(query, key, value, attention_mask, head_mask)
  File "/home/vyomkeshj/.conda/envs/ml_test/lib/python3.10/site-packages/transformers/models/gptj/modeling_gptj.py", line 150, in _attn
    attn_weights = torch.matmul(query, key.transpose(-1, -2))
RuntimeError: CUDA out of memory. Tried to allocate 24.00 MiB (GPU 0; 15.78 GiB total capacity; 13.30 GiB already allocated; 4.19 MiB free; 13.95 GiB reserved in total by PyTorch) If reserved memory is >> allocated memory try setting max_split_size_mb to avoid fragmentation.  See documentation for Memory Management and PYTORCH_CUDA_ALLOC_CONF